v2/tools/generator/internal/config/type_transformer.go

```diff
@@ -55,37 +56,163 @@ type TypeTransformer struct {
 	matchedProperties map[astmodel.TypeName]string
 }
 
-func produceTargetType(
-	target TransformTarget,
+func (target *TransformTarget) AppliesToType(t astmodel.Type) bool {
+	if target.appliesCache == nil {
+		target.appliesCache = make(map[astmodel.Type]bool)
+	}
+
+	if result, ok := target.appliesCache[t]; ok {
+		return result
+	}
+
+	result := target.appliesToType(t)
+	target.appliesCache[t] = result
+	return result
+}
+
+func (target *TransformTarget) appliesToType(t astmodel.Type) bool {
+	if target == nil {
+		return true
+	}
+
+	inspect := t
+	if target.Optional {
+		// Need optional
+		opt, ok := astmodel.AsOptionalType(inspect)
+		if !ok {
+			// but don't have optional
+			return false
```

Comment:
This could be a bit weird though, because what error is going to result when you set `Optional: True`, then match a non-optional type. Is it going to say "I couldn't find that property"? That's not actually correct though because it can find it, it just didn't match the optional requirement.

Reply:
I'll tweak the error message to say "matched types but all properties were excluded **by name or type**" (the bold bit is new)